tests/gamma/csm_observability_test.py

```diff
@@ -45,39 +92,383 @@ def setUpClass(cls):
+        with self.subTest("7_check_resource_label_keys"):
+            # all metrics should have the same set of monitored resource labels
+            # which come from the GMP job
+            for metric in _ALL_METRICS:
+                resource_labels = all_results[metric][0].resource.labels
+                for label_key in [
+                    "cluster",
+                    "instance",
+                    "job",
+                    "location",
+                    "namespace",
+                    "project_id",
+                ]:
+                    self.assertIn(label_key, resource_labels)
+
+        # Testing the values of metric labels
+        with self.subTest("8_check_metrics_label_values"):
+            for metric, time_series in all_results.items():
+                series = time_series[0]
+                self.assertEqual(metric, series.metric.type)
+                for label_key, label_value in series.metric.labels.items():
+                    if label_key == "pod" and self._is_client_metric(metric):
+                        # client pod name
+                        self.assertEqual(test_client.hostname, label_value)
+                    elif label_key == "pod" and self._is_server_metric(metric):
+                        # server pod name
+                        self.assertEqual(test_server.hostname, label_value)
+                    elif label_key == "grpc_method":
+                        self.assertEqual(_GRPC_METHOD_NAME, label_value)
+                    elif label_key == "grpc_target":
+                        # server namespace should be somewhere in the xds
+                        # server target
+                        self.assertIn(self.server_namespace, label_value)
+                    elif (
+                        label_key == "csm_remote_workload_canonical_service"
+                        and self._is_client_metric(metric)
+                    ):
+                        # the remote workload canonical service name
+                        # for the client is the one we set in the server
+                        self.assertEqual(
+                            _CSM_CANONICAL_SERVICE_NAME_SERVER, label_value
+                        )
+                    elif (
+                        label_key == "csm_remote_workload_canonical_service"
+                        and self._is_server_metric(metric)
+                    ):
+                        # the remote workload canonical service name
+                        # for the server is the one we set in the client
+                        self.assertEqual(
+                            _CSM_CANONICAL_SERVICE_NAME_CLIENT, label_value
+                        )
+                    elif label_key == "csm_remote_workload_cluster_name":
+                        # the cluster name should be somewhere in the
+                        # kube context
+                        self.assertIn(
+                            label_value, xds_k8s_flags.KUBE_CONTEXT.value
```

Comment:
Bad idea. There isn't no relation between the name of the cluster, and the name of the context. It just happens that gcloud generate context name in a certain way. But many choose to rename the context. In fact, my local run fail because of it:

```
E0126 16:05:28.531121 140704460725120 base_testcase.py:42] ----- PSM Test Case FAILED: CsmObservabilityTest.test_csm_observability -----
E0126 16:05:28.531229 140704460725120 base_testcase.py:95] (FAILURE) PSM Interop Test Failed: CsmObservabilityTest.test_csm_observability
^^^^^
[CsmObservabilityTest.test_csm_observability] PSM Failed Test Traceback BEGIN
Traceback (most recent call last):
  File "/Users/sergiitk/Development/psm-interop/tests/gamma/csm_observability_test.py", line 341, in test_csm_observability
    self.assertIn(
AssertionError: 'us-west1-a' not found in 'sergii-gamma'
[CsmObservabilityTest.test_csm_observability] PSM Failed Test Traceback END
```

Reply:
I added a sentinal value `VALUE_NOT_CHECKED` to indicate that we _do_ want to check whether the key exists, but we don't care about the value. This includes cluster_name, location, etc, for now.